src/local_manifest.rs

```diff
@@ -631,29 +632,27 @@ impl LocalFolderManifest {
     fn evolve_children_and_mark_updated(
         &self,
         data: HashMap<EntryName, Option<EntryID>>,
-        prevent_sync_pattern: &PyAny,
+        prevent_sync_pattern: &Regex,
         timestamp: DateTime,
     ) -> PyResult<Self> {
         let data = data
             .into_iter()
             .map(|(en, ei)| (en.0, ei.map(|ei| ei.0)))
             .collect();
-        let prevent_sync_pattern = py_to_rs_regex(prevent_sync_pattern)?;
         Ok(Self(self.0.clone().evolve_children_and_mark_updated(
             data,
-            &prevent_sync_pattern,
+            &prevent_sync_pattern.0 .0,
```

Comment:
Wrong search and replace ?

```suggestion
            &prevent_sync_pattern.0,
```

Reply:
This function expects a `::regex::regex` struct to we need to access the inner object twice here. Otherwise we would get an error like this
```
expected `struct regex::Regex`, found struct `libparsec::libparsec_types::regex::Regex
```

Reply:
Note that the extra space between the `0` and the `.` is added by `rustfmt`. I guess it the rust's coding style

Reply:
I don't like the `.0 .0`, Why not passing the Wrapped regex (`libparsec::Regex`) to it ?